readme.md

```diff
@@ -23,34 +23,124 @@ For global cli use:
 npm install -g firmata
 ```
 
-# Usage
+
+# REPL
+
+If you run *firmata* from the command line it will prompt you for the serial port. Then it will present you with a REPL with a board variable available.
+
+# Basic Usage
+
+### Using the `"ready"` event...
+
+#### With a path string: 
 
 ```js
 var Board = require("firmata");
-var board = new Board("path to serialport", function() {
+var board = new Board("system path or name");
+
+board.on("ready", () => {
   // Arduino is ready to communicate
 });
 ```
 
-Or
+#### With a Serialport object:
 
 ```js
+var Serialport = require("serialport");
 var Board = require("firmata");
-var board = new Board("path to serialport");
+var board = new Board(new Serialport(...));
 
-board.on("ready", function() {
+board.on("ready", () => {
   // Arduino is ready to communicate
 });
 ```
 
-# REPL
+#### With an Etherport object:
```

Comment:
May be useful to add that Etherport is a TCP server that can be used with StandardFirmataEthernet or StandardFirmataWiFi when configured as a TCP client (StandardFirmataEthernet can currently only be configured as a TCP client). There is also [etherport-client](https://github.com/mwittig/etherport-client) which is a TCP client that can be used with StandardFirmataWiFi when configured as a TCP server.